tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/bindgen/BindgenTestCase.java

```diff
@@ -0,0 +1,115 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.bindgen;
+
+import org.ballerinalang.test.BaseTest;
+import org.ballerinalang.test.context.BMainInstance;
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.ballerinalang.test.context.LogLeecher;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import static org.ballerinalang.test.context.LogLeecher.LeecherType.ERROR;
+import static org.ballerinalang.test.packaging.PackerinaTestUtils.deleteFiles;
+
+/**
+ * Integration tests for the bindgen tool.
+ *
+ * @since 1.2.5
+ */
+public class BindgenTestCase extends BaseTest {
+
+    private Path tempProjectsDirectory;
+    private Path projectPath;
+    private BMainInstance balClient;
+
+    @BeforeClass()
+    public void setUp() throws IOException, BallerinaTestException {
+        tempProjectsDirectory = Files.createTempDirectory("bal-test-integration-bindgen-");
+        // copy TestProject to a temp
+        Path originalTestProj1 = Paths.get("src", "test", "resources", "bindgen", "TestProject")
+                .toAbsolutePath();
+        projectPath = tempProjectsDirectory.resolve("module1");
+        copyFolder(originalTestProj1, projectPath);
+        balClient = new BMainInstance(balServer);
+    }
+
+    /**
+     * Generate Ballerina bindings covering multiple scenarios and build the project.
+     *
+     * @throws BallerinaTestException Error when executing the commands.
+     */
+    @Test(description = "Test the bindgen command functionality.")
+    public void bindgenTest() throws BallerinaTestException {
+        String buildMsg = "target/bin/module1.jar";
+        LogLeecher buildLeecher = new LogLeecher(buildMsg);
+
+        String bindgenMsg = "class could not be generated.";
+        LogLeecher bindgenLeecher = new LogLeecher(bindgenMsg, ERROR);
+
+        String runMsg1 = "Array util tests successful.";
+        String runMsg2 = "Exception handling tests successful.";
+        LogLeecher runLeecher1 = new LogLeecher(runMsg1);
+        LogLeecher runLeecher2 = new LogLeecher(runMsg2);
+
+        String[] args = {"-mvn=org.yaml:snakeyaml:1.25", "-o=./src/module1", "org.yaml.snakeyaml.Yaml",
+                "java.lang.Object", "java.lang.String", "java.lang.Class", "java.lang.StringBuffer",
```

Comment:
Why do we pass the full list of classes. In a normal case wouldn't it be few

Reply:
Yes, this is to make sure the compilation is successful for different scenarios and some of the most frequently used classes in the standard Java library. 

- `org.yaml.snakeyaml.Yaml` - build should pass only if the maven resolving is successful

- `java.lang.Object`, `java.lang.String`, `java.lang.Class`, `java.lang.StringBuffer`, `java.lang.Character`, `java.lang.Short`, `java.io.File`, `java.io.FileInputStream`, `java.io.OutputStream`, `java.io.FileOutputStream`, `java.util.HashSet`  - common used classes

- `java.lang.ArithmeticException` - generating an exception class

- `java.lang.AssertionError` - generating an error class

- `java.lang.StackTraceElement` - used to call a method that uses an array util function

- `java.lang.Character$Subset` - generating bindings for inner classes